Lib/test/test_socket.py

```diff
@@ -2277,6 +2272,26 @@ def testFromFd(self):
     def _testFromFd(self):
         self.serv_conn.send(MSG)
 
+    @requireAttrs(socket.socket, 'send_fds')
+    def testSendAndRecvFds(self):
+        fds = []
+        for i in range(1):
+            fd, path = tempfile.mkstemp()
+            self.addCleanup(os.unlink, path)
+            self.addCleanup(os.close, fd)
+            os.write(fd, str(i).encode())
+            fds.append(fd)
+        f = self.cli_conn.detach()
+        sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM, fileno=f)
+        self.addCleanup(sock.close)
+        with sock:
+            socket.send_fds(sock, MSG, fds)
+            msg, fds_list = socket.recv_fds(sock, len(MSG), 1024)
+            self.assertEqual(msg, MSG)
+
+    def _testSendAndRecvFds(self):
```

Comment:
What is this function for?

Reply:
This how all tests in this file are writtten. After writing the main test. I think You have to write this method to send any data that your test uses.

Without this function, running your tests fail with this:

======================================================================
ERROR: testSendAndRecvFds (test.test_socket.BasicTCPTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/captain/projects/cpython/Lib/test/test_socket.py", line 315, in _setUp
    test_method = getattr(self, '_' + methodname)
AttributeError: 'BasicTCPTest' object has no attribute '_testSendAndRecvFds'

======================================================================
ERROR: testSendAndRecvFds (test.test_socket.BasicTCPTest2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/captain/projects/cpython/Lib/test/test_socket.py", line 315, in _setUp
    test_method = getattr(self, '_' + methodname)
AttributeError: 'BasicTCPTest2' object has no attribute '_testSendAndRecvFds'

---------------------------------------------------------------------